.github/workflows/publish-package.yml

```diff
@@ -36,11 +45,25 @@ jobs:
       - name: Build
         run: yarn build
 
-      # TODO: we still need to make the tests work in CI environments
-      # will come back to this
-      # - name: Test
-      #   run: yarn test
+      - name: E2E tests
+        id: e2e
+        env:
+          API_KEY: ${{ secrets.API_KEY }}
+          OWNER_MNEMONIC: ${{ secrets.OWNER_MNEMONIC }}
+          PAYMASTER_POLICY_ID: ${{ secrets.PAYMASTER_POLICY_ID }}
+        run: yarn test:e2e
+
+      - name: Artifact of E2E tests
+        id: artifact
+        if: steps.e2e.outcome == 'success'
+        run: echo 'e2e=true' >> "$GITHUB_OUTPUT"
 
-      # Lerna publish will, compute the new version, update files, run precommit hooks, tag, publish, and push change log
+  # Lerna publish will, compute the new version, update files, run precommit hooks, tag, publish, and push change log
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: build_and_test
+    if: inputs.publish && (needs.build_and_test.outputs.artifact == 'true')
```

Comment:
from docs: "you may omit the ${{ }} expression syntax because GitHub Actions automatically evaluates the if conditional as an expression. Using the ${{ }} expression syntax turns the contents into a string, and strings are truthy."

I also check against the stringified boolean because the artifact variable `e2e` I believe is still a string.